src/mbgl/map/transform_state.cpp

```diff
@@ -321,16 +330,18 @@ void TransformState::moveLatLng(const LatLng& latLng, const ScreenCoordinate& an
 }
 
 void TransformState::setLatLngZoom(const LatLng &latLng, double zoom) {
+    const LatLng constrained = bounds.constrain(latLng);
+
     double newScale = zoomScale(zoom);
     const double newWorldSize = newScale * util::tileSize;
     Bc = newWorldSize / util::DEGREES_MAX;
     Cc = newWorldSize / util::M2PI;
 
     const double m = 1 - 1e-15;
-    const double f = util::clamp(std::sin(util::DEG2RAD * latLng.latitude), -m, m);
+    const double f = util::clamp(std::sin(util::DEG2RAD * constrained.latitude), -m, m);
 
     ScreenCoordinate point = {
-        -latLng.longitude * Bc,
+        -constrained.longitude * Bc,
```

Comment:
That's where we constrain the received `LatLng` to the specified bounds. I preferred this to be done when inserting because I suppose we set `LatLng` values much less than we request them from `TransformState`.